src/coreclr/jit/simdashwintrinsiclistarm64.h

```diff
@@ -132,6 +132,7 @@ SIMD_AS_HWINTRINSIC_ID(VectorT128,  op_Inequality,
 SIMD_AS_HWINTRINSIC_ID(VectorT128,  op_Multiply,                                            2,         {NI_VectorT128_op_Multiply,                     NI_VectorT128_op_Multiply,                      NI_VectorT128_op_Multiply,                      NI_VectorT128_op_Multiply,                      NI_VectorT128_op_Multiply,                      NI_VectorT128_op_Multiply,                      NI_Illegal,                                     NI_Illegal,                                     NI_VectorT128_op_Multiply,                      NI_VectorT128_op_Multiply},                     SimdAsHWIntrinsicFlag::None)
 SIMD_AS_HWINTRINSIC_ID(VectorT128,  op_Subtraction,                                         2,         {NI_AdvSimd_Subtract,                           NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Subtract,                            NI_AdvSimd_Arm64_Subtract},                     SimdAsHWIntrinsicFlag::None)
 SIMD_AS_HWINTRINSIC_ID(VectorT128,  SquareRoot,                                             1,         {NI_Illegal,                                    NI_Illegal,                                     NI_Illegal,                                     NI_Illegal,                                     NI_Illegal,                                     NI_Illegal,                                     NI_Illegal,                                     NI_Illegal,                                     NI_AdvSimd_Arm64_Sqrt,                          NI_AdvSimd_Arm64_Sqrt},                         SimdAsHWIntrinsicFlag::None)
+SIMD_AS_HWINTRINSIC_ID(VectorT128,  Sum,                                                    1,         {NI_AdvSimd_Arm64_AddAcross,                    NI_AdvSimd_Arm64_AddAcross,                     NI_AdvSimd_Arm64_AddAcross,                     NI_AdvSimd_Arm64_AddAcross,                     NI_AdvSimd_Arm64_AddAcross,                     NI_AdvSimd_Arm64_AddAcross,                     NI_VectorT128_Sum,                              NI_VectorT128_Sum,                              NI_VectorT128_Sum,                              NI_VectorT128_Sum},                             SimdAsHWIntrinsicFlag::None)
```

Comment:
The assert looks to come because you have these directly calling `Arm64.AddAcross`, where `Vector<T>Sum` returns `T` but `Arm64.AddAcross` returns `Vector64<T>`.

This needs to be handled specially with a corresponding `NI_Vector64_ToScalar` to work, afaict.